controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -365,29 +373,74 @@ func (r *ReplicationGroupReconciler) processSnapshotEvent(ctx context.Context, g
 		return err
 	}
 
-	if _, err := remoteClient.GetSnapshotClass(ctx, actionAnnotation.SnapshotClass); err != nil {
-		log.Error(err, "Snapshot class does not exist on remote cluster. Not creating the remote snapshots.")
-		return err
-	}
+	namespace := actionAnnotation.SnapshotNamespace
 
-	if _, err := remoteClient.GetNamespace(ctx, actionAnnotation.SnapshotNamespace); err != nil {
-		log.V(common.InfoLevel).Info("Namespace - " + actionAnnotation.SnapshotNamespace + " not found, creating it.")
-		nsRef := makeNamespaceReference(actionAnnotation.SnapshotNamespace)
-
-		err = remoteClient.CreateNamespace(ctx, nsRef)
+	if _, err := remoteClient.GetNamespace(ctx, namespace); err != nil {
+		log.V(common.InfoLevel).Info("Namespace - " + namespace + " not found, creating it.")
+		err = createNamespace(ctx, namespace, remoteClient)
 		if err != nil {
-			msg := "unable to create the desired namespace" + actionAnnotation.SnapshotNamespace
-			log.V(common.ErrorLevel).Error(err, msg)
 			return err
```

Comment:
Can we mention specific error here, like **return fmt.Errorf("unable to create desired namespace: %s", err.Error())**  ??